### ISSUE TITLE
Aliases don't seem to be working via .unimportedrc.json

### ISSUE THREAD

**Issue by multipolygon** [smeijer/unimported#45]
Otherwise an excellent tool, thank you

**Comment by multipolygon** [comment]
Just experimented and found that it was working in `1.11.0` but not `1.12.0` - thanks!

**Comment by smeijer** [comment]
Can you provide a repo? Or maybe post your config and a few example imports?

**Comment by mzalevski** [comment]
@smeijer I had the same problem z the @multipolygon and the `1.11.0` worked for me as well.

I am using React 17 CRA app and the `import` statements look like this: `import Chart from "components/Chart";` where the `components` folder is in `src` folder.

Here is my config:
```
{
  "aliases": { "*": "src/*" }, < - - - - - - I've added this line
  "ignoreUnresolved": [],
  "ignoreUnimported": [],
  "ignoreUnused": [],
  "ignorePatterns": [
    "**/node_modules/**",
    "**/*.stories.{js,jsx,ts,tsx}",
    "**/*.tests.{js,jsx,ts,tsx}",
    "**/*.test.{js,jsx,ts,tsx}",
    "**/*.spec.{js,jsx,ts,tsx}",
    "**/tests/**",
    "**/__tests__/**",
    "**/*.d.ts"
  ]
}

```

**Comment by smeijer** [comment]
I've made a small repo using the following script:

```shell
npx create-react-app my-app
cd my-app
echo '{ "aliases": { "*": "src/*" } }' > .unimportedrc.json
echo 'import Chart from "components/Chart";' >> src/index.js
mkdir src/components
touch src/components/Chart.js
touch src/components/Chart2.js
```

Then I ran `npx unimported` in `my-app` (next to `package.json`), and the output was as expected:

```shell
unimported-repo/my-app on  main [✘!?] is 📦 v0.1.0 
➜ npx unimported

       summary               unimported v1.12.1
──────────────────────────────────────────────────────────
       entry file          : src/index.js

       unresolved imports  : 0
       unused dependencies : 4
       unimported files    : 2


─────┬────────────────────────────────────────────────────
     │ 4 unused dependencies
─────┼────────────────────────────────────────────────────
   1 │ @testing-library/jest-dom
   2 │ @testing-library/react
   3 │ @testing-library/user-event
   4 │ react-scripts
─────┴────────────────────────────────────────────────────


─────┬────────────────────────────────────────────────────
     │ 2 unimported files
─────┼────────────────────────────────────────────────────
   1 │ src/components/Chart2.js
   2 │ src/setupTests.js
─────┴────────────────────────────────────────────────────
```

I'm happy to take a look at a repo, but without one, I can only conclude that it works as expected.

**Comment by smeijer** [comment]
Gonna close this for now. Feel free to reopen with a repo.

**Comment by iest** [comment]
I'm afraid I've gotta +1 on this — I was also having issues with aliases just not working no matter what the configuration. Then I `npm i -g unimported@1.11.0` and ran it without issue.

Sorry I can't provide a repro, but I can provide some details:

1. Pure JS project (no TS)
2. We have an alias file that looks like the below, and is pulled into our `.eslintrc` to work with `eslint-import-resolver-babel-module`, and into our `babel.config.js` to work with `babel-plugin-module-resolver`:
  ```js
  module.exports = {
    '@helpers': './src/helpers',
    '@actions': './src/actions',
    '@styles': './src/styles',
    '@messaging': './src/messaging',
    '@common': './src/components/common',
    '@components': './src/components',
    '@backend': './src/Backend',
    '@images': './src/img',
    '@src': './src',
    '@test': './src/test',
  };
  ```
3. The `.unimportedrc.json` had the following in it:
  ```js
  "aliases": {
    "@helpers/*": ["src/helpers", "src/helpers/*"],
    "@actions/*": ["src/actions", "src/actions/*"],
    "@styles/*": ["src/styles", "src/styles/*"],
    "@messaging/*": ["src/messaging", "src/messaging/*"],
    "@common/*": ["src/components/common", "src/components/common/*"],
    "@components/*": ["src/components", "src/components/*"],
    "@backend": ["src/Backend"],
    "@images/*": ["src/img", "src/img/*"],
    "@src/*": ["src", "src/*"],
    "@test/*": ["src/test", "src/test/*"]
  }
  ```

tl;dr
- `$ npx unimported` didn't work
- `$ npm i -g unimported@1.11.0` and `$ unimported` worked

**Comment by smeijer** [comment]
I do believe that you're experiencing issues, but I cannot help without a reproduction.

When I try to build a reproduction based on the info that you now provide, I cannot make unimported come up with false results.

What I've tried:

**package.json**
```json
{
  "main": "src/index.js"
}
```

**.unimportedrc.json**
```json
{
  "aliases": {
    "@helpers/*": ["src/helpers", "src/helpers/*"],
  }
}
```

**src/index.js**
```js
import { random } from '@helpers/random';
```

**src/helpers/add.js**
```js
export function add() {}
```

**src/helpers/random.js**
```js
export function random() {}
```

Unimported lists only `src/helpers.add.js` as unimported, which is the expected result.

Making a reproduction like this takes 5 minutes. If you need to have your project supported, please take a few minutes to create a reproduction.

**Comment by smeijer** [comment]
@iest, is it by any chance all index files that are incorrectly reported?

**Comment by iest** [comment]
@smeijer not all, but certainly one: the only place we had an index file was `@actions`, and we import both from `@actions` as well as `@actions/foobar`...

**Comment by smeijer** [comment]
Ok. Because what I see now is that `import x from '@actions'` results in a false positive for `@actions/index`. I'll give it a shot to fix that. But `@actions/foobar` should just work.

**Comment by smeijer** [comment]
And the other cases. Are those by any chance aliases pointing to files instead of directories?

**Comment by iest** [comment]
In some cases yes — the aliases point to files

**Comment by smeijer** [comment]
I've published `v14`. Can you give that a shot and let me know if that works better?

**Comment by iest** [comment]
@smeijer it works perfectly! Excellent, excellent work.

I owe you a cold one 🍻
